### PR TITLE
[PlSql] Support of alter operator

### PR DESCRIPTION
**Pull request by jiangtao69039**
https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/ALTER-OPERATOR.html#GUID-F00A0AC8-36C8-4EAC-A9BB-B3D42C5EEEDE

Add support of alter operator statement.

### CODE REVIEW

File: sql/plsql/PlSqlParser.g4 [hunk 1]
```diff
@@ -2098,6 +2107,40 @@ create_zonemap_as_subquery
     : AS subquery
     ;
 
+alter_operator
+    : ALTER OPERATOR operator_name (add_binding_clause | drop_binding_clause | COMPILE)
+    ;
+
+add_binding_clause
+    : ADD BINDING LEFT_PAREN (datatype (COMMA datatype)*) RIGHT_PAREN
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:2115]: Please remove redundant parentheses:

```suggestion
    : ADD BINDING LEFT_PAREN datatype (COMMA datatype)* RIGHT_PAREN
```

File: sql/plsql/PlSqlParser.g4 [hunk 2]
```diff
@@ -2098,6 +2107,40 @@ create_zonemap_as_subquery
...
+        RETURN LEFT_PAREN? datatype RIGHT_PAREN? implementation_clause? using_function_clause
+    ;
+
+implementation_clause
+    : ANCILLARY TO primary_operator_list | operator_context_clause
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:2120]: Fix formatting. Every first-level alternative should be placed on the new line:

```suggestion
    : ANCILLARY TO primary_operator_list
    | operator_context_clause
```

File: sql/plsql/PlSqlParser.g4 [hunk 3]
```diff
@@ -2098,6 +2107,40 @@ create_zonemap_as_subquery
...
+    ;
+
+primary_operator_list
+    : primary_operator_item (COMMA primary_operator_item)*
+    ;
+
+primary_operator_item
+    : schema_object_name LEFT_PAREN (datatype (COMMA datatype)*) RIGHT_PAREN
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:2128]:
```suggestion
    : schema_object_name LEFT_PAREN datatype (COMMA datatype)* RIGHT_PAREN
```

File: sql/plsql/PlSqlParser.g4 [hunk 4]
```diff
@@ -2098,6 +2107,40 @@ create_zonemap_as_subquery
...
+    : WITH INDEX CONTEXT COMMA SCAN CONTEXT implementation_type_name (COMPUTE ANCILLARY DATA)?
+        (WITH COLUMN CONTEXT)?
+    ;
+
+using_function_clause
+    : USING operator_function_name
+    ;
+
+drop_binding_clause
+    : DROP BINDING LEFT_PAREN (datatype (COMMA datatype)*) RIGHT_PAREN FORCE?
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:2141]:
```suggestion
    : DROP BINDING LEFT_PAREN datatype (COMMA datatype)* RIGHT_PAREN FORCE?
```

File: sql/plsql/examples/alter_operator.sql [hunk 5]
```diff
@@ -0,0 +1 @@
+ALTER OPERATOR eq_op COMPILE;
```
Comment (KvanTTT) [inline sql/plsql/examples/alter_operator.sql:1]: Maybe it makes sense to add more examples on the new syntax. But it's optionally.

**Review by KvanTTT — CHANGES_REQUESTED** [review]

File: sql/plsql/examples/alter_operator.sql (hunk 5 above)
Comment (jiangtao69039) [reply sql/plsql/examples/alter_operator.sql:1]: I only found this one example from the official documentation.

**Review by KvanTTT — APPROVED** [review]

**Pull request merged**